utils/git_secrets.py

```diff
@@ -0,0 +1,65 @@
+import os
+import tempfile
+import shutil
+import logging
+
+from utils.defer import defer
+
+from git import Repo
+from os import path
+from subprocess import PIPE, Popen
+
+
+@defer
+def scan_history(repo_url, existing_keys, defer=None):
+    logging.info('scanning {}'.format(repo_url))
+    wd = tempfile.mkdtemp()
+    defer(lambda: shutil.rmtree(wd))
+
+    repo = Repo.clone_from(repo_url, wd)
+    DEVNULL = open(os.devnull, 'w')
+    proc = Popen(['git', 'secrets', '--register-aws'],
+                 cwd=wd, stdout=DEVNULL)
+    proc.communicate()
+    proc = Popen(['git', 'secrets', '--scan-history'],
+                 cwd=wd, stdout=PIPE, stderr=PIPE)
+    _, err = proc.communicate()
+    if proc.returncode == 0:
+        return []
+
+    logging.info('found suspects in {}'.format(repo_url))
+    suspcted_files = get_suspected_files(err)
+    leaked_keys = get_leaked_keys(repo, suspcted_files, existing_keys)
+    if leaked_keys:
+        logging.info('found suspected leaked keys: {}'.format(leaked_keys))
+
+    return leaked_keys
+
+
+def get_suspected_files(error):
```

Comment:
can you drop an example of the output so we can understand that parsing?

Reply:
```
$ git-secrets --scan-history
9e43530e337d7d0feec726db13d422a78939b183:training/datastore/get_preprocess_data.py:33:                 aws_secret_access_key='2+Po/7pyFzSNnHFfAQDbLpZ6p/0uHE5Ujx7mE1W7',

[ERROR] Matched one or more prohibited patterns

Possible mitigations:
- Mark false positives as allowed using: git config --add secrets.allowed ...
- Mark false positives as allowed by adding regular expressions to .gitallowed at repository's root directory
- List your configured patterns: git config --get-all secrets.patterns
- List your configured allowed patterns: git config --get-all secrets.allowed
- List your configured allowed patterns in .gitallowed at repository's root directory
- Use --no-verify if this is a one-time false positive
```